### PR TITLE
Generic type map

### PR DESCRIPTION
**Pull request by tsvisabo**
- Implement generic type map (fixed hash size, do we want to support different sizes?)

### CODE REVIEW

**Review by thibault-martinez — COMMENTED** [review]
Do we still need the previous `hash_int_map_t` then ?

File: common/storage/sql/sqlite3/tests/test_sqlite3.c [hunk 1]
```diff
@@ -300,11 +300,11 @@ void test_milestone_state_delta(void) {
   flex_trit_t *hashed_hash;
 
   flex_trits_from_trits(hash, HASH_LENGTH, trits, HASH_LENGTH, HASH_LENGTH);
-  for (int i = -1000; i <= 1000; i++) {
+  for (int64_t i = -1000; i <= 1000; i++) {
     hashed_hash = iota_flex_digest(hash, HASH_LENGTH);
     memcpy(hash, hashed_hash, FLEX_TRIT_SIZE_243);
     free(hashed_hash);
-    TEST_ASSERT(state_delta_add(&state_delta1, hash, i) == RC_OK);
+    TEST_ASSERT(hash_to_int64_t_map_add(&state_delta1, hash, i) == RC_OK);
```
Comment (thibault-martinez) [inline common/storage/sql/sqlite3/tests/test_sqlite3.c]: Why can't you use `state_delta_add` here ?

File: consensus/snapshot/state_delta.c [hunk 2]
```diff
@@ -42,7 +42,7 @@ retcode_t state_delta_add_or_replace(state_delta_t *const state,
   if (entry) {
     entry->value = value;
   } else {
-    if ((ret = state_delta_add(state, hash, value)) != RC_OK) {
+    if ((ret = hash_to_int64_t_map_add(state, hash, value)) != RC_OK) {
```
Comment (thibault-martinez) [inline consensus/snapshot/state_delta.c]: Why can't you use state_delta_add here ?

File: consensus/snapshot/state_delta.c [hunk 3]
```diff
@@ -132,18 +132,16 @@ retcode_t state_delta_serialize(state_delta_t const *const delta,
...
 retcode_t state_delta_deserialize(byte_t const *const bytes, size_t const size,
                                   state_delta_t *const delta) {
+  retcode_t ret;
   uint64_t offset = 0;
   state_delta_entry_t *new = NULL;
-
   for (size_t i = 0; i < size / (FLEX_TRIT_SIZE_243 + sizeof(int64_t)); i++) {
-    if ((new = malloc(sizeof(state_delta_entry_t))) == NULL) {
-      return RC_SNAPSHOT_OOM;
+    if ((ret = hash_to_int64_t_map_add(
```
Comment (thibault-martinez) [inline consensus/snapshot/state_delta.c]: `state_delta_add` ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c [hunk 4]
```diff
@@ -163,12 +163,13 @@ void test_cw_gen_topology(test_tangle_topology topology) {
   }
   // First Entry, for both topologies, has to equal the number of total
   // approvers + own weight
-  TEST_ASSERT_EQUAL_INT(out.cw_ratings->value, num_approvers + 1);
+  // TEST_ASSERT_EQUAL_INT64(out.cw_ratings->value, num_approvers + 1);
```
Comment (thibault-martinez) [inline consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Can we uncomment ?

**Review by thibault-martinez — CHANGES_REQUESTED** [review]

File: consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c (hunk 4 above)
Comment (tsvisabo) [reply consensus/exit_probability_randomizer/tests/test_exit_probability_randomizer.c]: Thanks

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**